be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -392,72 +392,73 @@ bool PipelineDriver::_check_fragment_is_canceled(RuntimeState* runtime_state) {
     if (op_state >= OperatorStage::FINISHING) {
-        return;
+        return Status::OK();
     }
 
     VLOG_ROW << strings::Substitute("[Driver] finishing operator [driver=$0] [operator=$1]", to_readable_string(),
                                     op->get_name());
     {
         SCOPED_TIMER(op->_finishing_timer);
-        op->set_finishing(state);
+        op_state = OperatorStage::FINISHING;
+        return op->set_finishing(state);
     }
-    op_state = OperatorStage::FINISHING;
 }
 
-void PipelineDriver::_mark_operator_finished(OperatorPtr& op, RuntimeState* state) {
-    _mark_operator_finishing(op, state);
+Status PipelineDriver::_mark_operator_finished(OperatorPtr& op, RuntimeState* state) {
+    RETURN_IF_ERROR(_mark_operator_finishing(op, state));
     auto& op_state = _operator_stages[op->get_id()];
     if (op_state >= OperatorStage::FINISHED) {
-        return;
+        return Status::OK();
     }
 
     VLOG_ROW << strings::Substitute("[Driver] finished operator [driver=$0] [operator=$1]", to_readable_string(),
                                     op->get_name());
     {
         SCOPED_TIMER(op->_finished_timer);
-        op->set_finished(state);
+        op_state = OperatorStage::FINISHED;
+        return op->set_finished(state);
     }
-    op_state = OperatorStage::FINISHED;
 }
 
-void PipelineDriver::_mark_operator_cancelled(OperatorPtr& op, RuntimeState* state) {
-    _mark_operator_finished(op, state);
+Status PipelineDriver::_mark_operator_cancelled(OperatorPtr& op, RuntimeState* state) {
+    RETURN_IF_ERROR(_mark_operator_finished(op, state));
     auto& op_state = _operator_stages[op->get_id()];
     if (op_state >= OperatorStage::CANCELLED) {
-        return;
+        return Status::OK();
     }
 
     VLOG_ROW << strings::Substitute("[Driver] cancelled operator [driver=$0] [operator=$1]", to_readable_string(),
                                     op->get_name());
-    op->set_cancelled(state);
     op_state = OperatorStage::CANCELLED;
+    return op->set_cancelled(state);
 }
 
-void PipelineDriver::_mark_operator_closed(OperatorPtr& op, RuntimeState* state) {
+Status PipelineDriver::_mark_operator_closed(OperatorPtr& op, RuntimeState* state) {
     if (_fragment_ctx->is_canceled()) {
-        _mark_operator_cancelled(op, state);
+        RETURN_IF_ERROR(_mark_operator_cancelled(op, state));
     } else {
-        _mark_operator_finished(op, state);
+        RETURN_IF_ERROR(_mark_operator_finished(op, state));
     }
 
     auto& op_state = _operator_stages[op->get_id()];
     if (op_state >= OperatorStage::CLOSED) {
-        return;
+        return Status::OK();
     }
 
     VLOG_ROW << strings::Substitute("[Driver] close operator [driver=$0] [operator=$1]", to_readable_string(),
                                     op->get_name());
     {
         SCOPED_TIMER(op->_close_timer);
-        op->close(state);
+        op_state = OperatorStage::CLOSED;
+        RETURN_IF_ERROR(op->close(state));
```

Comment:
If `op->close` returns error, `close()` of the operators after `op` cannot be called. Will this cause memory leak?